src/lib/qnio/utils.c

```diff
@@ -9,6 +9,117 @@
  */
 
 #include "defs.h"
+#include "qnio.h"
+
+int 
+is_secure()
+{
+    if (access(SECURE_IMPL, F_OK) != 0)
```

Comment:
QEMU aims to avoid guest config relying on external configuration files because that makes part of the setup invisible to the management tools. Instead all relevant parameters that influence operation of QEMU are explicitly configured via its command line. Compare with the RBD block device - while librbd has ability to use a config file in /etc/, for example to set passwords, QEMU will explicitly take the passwords on the command line instead of using the config file, so the mgmt app can be in complete control.

This allows the mgmt application to fully control QEMU operation, without having to rely on someone having manually edited a configuration file on the host previously. So QEMU should not have any dependancy on a /etc/vxhs.conf file - all aspects should be configurable via QEMU options, and thus exposed upto the libvirt management system, and in turn exposed to apps.

The libqnio API for starting a session would really be better refactored. It currently is

void *iio_open(const char *uri, const char *devid, uint32_t flags);

which both creates the handle and opens the session. I'd suggest this needs splitting. For example

void *iio_new();
int32_t iio_open(void *dev_handle, const char *uri, const char *devid, uint32_t flags);

Then you can add 

   iio_set_tls(void *dev_handle, const char *capath, const char *cacrl, const char *certpath, const char *keypath);

The application would call iio_set_tls after iio_new(), but before iio_open in order to explicitly request use of TLS on that handle.